thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -3005,7 +3005,10 @@ void CHashTableRowTable::init(rowidx_t sz)
     // reinitialize if need bigger or if requested size is much smaller than existing
     rowidx_t newMaxRows = activity.queryRowManager()->getExpectedCapacity(sz * sizeof(rowidx_t *), activity.allocFlags) / sizeof(rowidx_t *);
     if (newMaxRows <= maxRows && ((maxRows-newMaxRows) <= HASHDEDUP_HT_INC_SIZE))
+    {
+        clear();
         return;
+    }
     ReleaseThorRow(rows);
```

Comment:
You're right, I've added a clearRows() for this.
If there were any rows it would have meant they'd have leaked.
However, it's unlikely to have happened afiacs, because in the non child-query case, the buckets and HT's have been flushed/cleared and in the CQ case, the HT size _could_ grow but only if the input meta data changed (# of input rows) which is what determines the initial # of buckets/HT's and their size.